### ISSUE TITLE
Unable to access Variables with ACL

### ISSUE THREAD

**Issue by mr-karan** [hashicorp/nomad#15142]
### Nomad version
Output from `nomad version`

1.4.1

### Issue

I've created an ACL Policy to allow Job operators to "list" and "read" all secrets under the path `nomad/jobs`. However that ACL policy seems to have no effect as these secrets are still not visible to the user with a token from this policy.

### Reproduction steps

I've created an ACL Policy for job operators:

```
namespace "dev" {
  policy = "write"

  variables {
    path "nomad/jobs/*" {
      capabilities = ["read", "list"]
    }
  }
}


plugin {
  policy = "read"
}

node {
  policy = "read"
}

quota {
  policy = "read"
}
```

#### Expected Result

The above policy is for `dev` namespace and it should allow listing and reading all variables inside `nomad/jobs` path.

#### Actual Result

However, on opening the UI as a job operator with that policy, I only see this:

![image](https://user-images.githubusercontent.com/5689132/199915361-999b006c-05e5-48c3-97f3-a933119296eb.png)

v/s with a management token:

![image](https://user-images.githubusercontent.com/5689132/199915464-aee15ce2-b148-4cbf-8d68-069459f3c21b.png)

**Comment by jrasell** [comment]
Hi @mr-karan and thanks for raising this issue. Whilst I could reproduce this behaviour in v1.4.1, running the same test against v1.4.2 did not produce the same error. I believe this bug was fixed as part of the work conducted for https://github.com/hashicorp/nomad/issues/15012. Would you be able to test you reproduction with v1.4.2?

**Comment by mr-karan** [comment]
I can confirm this is fixed with 1.4.2.

Thanks a lot!
